packages/views.api.unstable/src/messages/hostMessages.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { AsyncParameterValueMap, ParameterConfig } from "../config.js";
+
+// Interfaces and type guards for messages passed from the host Foundry UI to the child view
+
+export interface HostParametersUpdatedMessage<CONFIG extends ParameterConfig> {
+  type: "host.update-parameters";
+  parameters: AsyncParameterValueMap<CONFIG>;
+}
+
+export type HostMessage<CONFIG extends ParameterConfig> =
```

Comment:
I assume you intend this to be a union at some point but I think we should reconsider for this type of project.

You are actually going to be running in a split world where the reality of the host will very likely be newer than the reality of the child. 

This means at some future point you will have the Host thinking there are 2 messages it sends and the client thinking there is only one. 

You may exhaust the union via if or switch and think you have handled everything when you in fact have not. The union will give you false confidence.

If we want to go this way, I think we need to be sure that adding to this union is a semver breaking change and that we communicate with the host which version of the API we are using. We probably need to do this no matter what actually.

Reply:
I either way we probably need to make a base type that is roughly: `{ type: string, payload: X }`

Reply:
yes, the intent is to be a union. i can update to use a (versioned?) base type 🤔

Reply:
Should we bake an unknown/unsupported event type concept into the union so that we are encouraged to be explicit on handling that case?